### PR TITLE
ZCS-3545:API to add user profile image and return it

### PR DESCRIPTION
**Pull request by sneha-patil-synacor**
API to add user profile image and return it in GetInfo and GetPrefs requests.
Profile image will be added to ldap binary attribute - zimbraPrefAccountProfileImage
User needs to upload the image first and then pass the aid as value in ModifyPrefs to add a profile image to the account.
Validation done to verify that its valid image file.
Image file is restricted upto 3MB.

Added Junit test.
ant test-all passed.

### CODE REVIEW

File: store/src/java/com/zimbra/cs/service/account/ModifyPrefs.java [hunk 1]
```diff
@@ -76,14 +87,62 @@ public Element handle(Element request, Map<String, Object> context) throws Servi
...
             int offset = ch == '+' || ch == '-' ? 1 : 0;
             if (!name.startsWith(PREF_PREFIX, offset))
-                throw ServiceException.INVALID_REQUEST("pref name must start with " + PREF_PREFIX, null);
+                throw ServiceException.INVALID_REQUEST("pref name must start with " + PREF_PREFIX,
+                    null);
 
-            AttributeInfo attrInfo = AttributeManager.getInstance().getAttributeInfo(name.substring(offset));
+            if (Provisioning.A_zimbraPrefAccountProfileImage.equals(name)) {
+                String attachId = value;
+                Upload up = FileUploadServlet.fetchUpload(zsc.getAuthtokenAccountId(), attachId,
```
Comment (rupalid) [inline store/src/java/com/zimbra/cs/service/account/ModifyPrefs.java]: Can we extract this to a utility method.

File: store/src/java/com/zimbra/cs/service/account/ModifyPrefs.java [hunk 2]
```diff
@@ -76,14 +87,62 @@ public Element handle(Element request, Map<String, Object> context) throws Servi
...
+                    contentType = MimeDetect.getMimeDetect().detect(attach.getInputStream());
+                } catch (IOException ioe) {
+                    throw MailServiceException.MESSAGE_PARSE_ERROR(ioe);
+                }
+                if (contentType == null
+                    || contentType.matches(MimeConstants.CT_IMAGE_WILD) == false) {
+                    throw MailServiceException
+                        .INVALID_IMAGE("Attached image is not a valid image file");
+                }
+                if (up.getSize() > 3145728l) {
```
Comment (rupalid) [inline store/src/java/com/zimbra/cs/service/account/ModifyPrefs.java]: make this number into a constant

File: store/src/java/com/zimbra/cs/service/account/ModifyPrefs.java (hunk 1 above)
Comment (sneha-patil-synacor) [reply store/src/java/com/zimbra/cs/service/account/ModifyPrefs.java]: Done

File: store/src/java/com/zimbra/cs/service/account/ModifyPrefs.java [hunk 3]
```diff
@@ -185,6 +200,45 @@ public static void sendEmailVerificationLink(Account authAccount, Account ownerA
         }
     }
 
+    private static String getImageBase64(ZimbraSoapContext zsc, String attachId)
```
Comment (rupalid) [inline store/src/java/com/zimbra/cs/service/account/ModifyPrefs.java]: Can you make it public static and move it to a new class FileUploadServletUtil in package com.zimbra.cs.service.util

Comment (sneha-patil-synacor) [reply store/src/java/com/zimbra/cs/service/account/ModifyPrefs.java]: Done

**Pull request closed without merging**